host/sgx/sgxsign.c

```diff
@@ -585,6 +587,45 @@ static oe_result_t _init_sigstruct(
     return result;
 }
 
+oe_result_t oe_sgx_sign_enclave_from_engine(
+    const OE_SHA256* mrenclave,
+    uint64_t attributes,
+    uint16_t product_id,
+    uint16_t security_version,
+    const char *engine_id,
+    const char *engine_load_path,
+    const char *key_id,
+    sgx_sigstruct_t* sigstruct)
+{
+    oe_rsa_private_key_t rsa;
+    bool rsa_initalized = false;
+    oe_result_t result = OE_UNEXPECTED;
+
+    if (sigstruct)
```

Comment:
You could move the memset down to line 610 and then you wouldn't need the IF test.  (The difference is just that if one of the other arguments is null and we fail, it won't be zeroed, but I think that's ok and more readable.)

Reply:
yup. done